output/cloud/expv2/output.go

```diff
@@ -275,14 +287,36 @@ func (o *Output) flushMetrics() {
 	o.logger.WithField("t", time.Since(start)).Debug("Successfully flushed buffered samples to the cloud")
 }
 
+func (o *Output) runFlushRequestMetadatas() {
+	t := time.NewTicker(o.config.TracesPushInterval.TimeDuration())
+
+	for i := int64(0); i < o.config.TracesPushConcurrency.Int64; i++ {
+		o.wg.Add(1)
+		go func() {
+			defer func() {
+				t.Stop()
+				o.wg.Done()
+			}()
+
+			for {
+				select {
+				case <-t.C:
+					o.flushRequestMetadatas()
+				case <-o.stop:
+					return
+				case <-o.abort:
+					return
+				}
+			}
+		}()
+	}
+}
+
 // flushRequestMetadatas periodically flushes traces collected in RequestMetadatasCollector using flusher.
 func (o *Output) flushRequestMetadatas() {
 	start := time.Now()
 
-	ctx, cancel := context.WithTimeout(context.Background(), o.config.TracesPushInterval.TimeDuration())
-	defer cancel()
-
-	err := o.requestMetadatasFlusher.flush(ctx)
+	err := o.requestMetadatasFlusher.flush(context.Background())
```

Comment:
I think we left this context so that this output can use it - if we are not going to use it - can we just drop it?

Reply:
Do you mean remove it from the `flusher interface` and all its implementations?

Reply:
I meant the context argument specifically.

But to be honest I am not really a fan of adding interfaces for the sake of adding interfaces or to make stuff .. mockable. Especially in cases like this where I feel the end result is a singel function call.

I do feel though that dropping the whole interface might be more invbolved and not to be part of this PR.

Reply:
After reading this [comment](https://raintank-corp.slack.com/archives/C042WNQFZEZ/p1688460082544839?thread_ts=1687937713.085209&cid=C042WNQFZEZ) in slack I removed the context, please let me know if now looks better to you.